Source/Request.swift

```diff
@@ -35,7 +35,16 @@ public class Request {
     // MARK: Properties
 
     /// The delegate for the underlying task.
-    public let delegate: TaskDelegate
+    public internal(set) var delegate: TaskDelegate {
+        get {
+            taskDelegateLock.lock() ; defer { taskDelegateLock.unlock() }
+            return taskDelegate
+        }
+        set {
+            taskDelegateLock.lock() ; defer { taskDelegateLock.unlock() }
+            taskDelegate = newValue
+        }
+    }
```

Comment:
If we decide this is unnecessary, then it's really simple. We just need to remove the `taskDelegate` and the `taskDelegateLock` and revert this change while maintaining the `internal(set) var delegate` change.